cmd/kubetoken/validator.go

```diff
@@ -0,0 +1,429 @@
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package kubetoken
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/url"
+	"sync"
+	"time"
+
+	oktetoLog "github.com/okteto/okteto/pkg/log"
+	"github.com/okteto/okteto/pkg/okteto"
+	"github.com/okteto/okteto/pkg/types"
+	authenticationv1 "k8s.io/api/authentication/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	clientcmdapi "k8s.io/client-go/tools/clientcmd/api"
+)
+
+const (
+	tokenRequestKind = "TokenRequest"
+)
+
+var (
+	errEmptyNamespace           = errors.New("namespace cannot be empty")
+	errEmptyContext             = errors.New("context name cannot be empty")
+	errTokenRequestNotSupported = errors.New("kubernetes cluster does not support TokenRequest")
+
+	valdationTimeout = 3 * time.Second
+)
+
+// validator is the interface that wraps the Validate method.
+type validator interface {
+	Validate(ctx context.Context) error
+}
+
+type preReqCfg struct {
+	ctxName string
+	ns      string
+
+	k8sClientProvider    okteto.K8sClientProvider
+	oktetoClientProvider types.OktetoClientProvider
+}
+
+type option func(*preReqCfg)
+
+func withCtxName(ctxName string) option {
+	return func(cfg *preReqCfg) {
+		cfg.ctxName = ctxName
+	}
+}
+
+func withNamespace(ns string) option {
+	return func(cfg *preReqCfg) {
+		cfg.ns = ns
+	}
+}
+
+func withK8sClientProvider(k8sClientProvider okteto.K8sClientProvider) option {
+	return func(cfg *preReqCfg) {
+		cfg.k8sClientProvider = k8sClientProvider
+	}
+}
+
+func withOktetoClientProvider(oktetoClientProvider types.OktetoClientProvider) option {
+	return func(cfg *preReqCfg) {
+		cfg.oktetoClientProvider = oktetoClientProvider
+	}
+}
+
+func defaultPreReqCfg() *preReqCfg {
+	return &preReqCfg{
+		k8sClientProvider:    okteto.NewK8sClientProvider(),
+		oktetoClientProvider: okteto.NewOktetoClientProvider(),
+	}
+}
+
+// preReqValidator validates that all the pre-reqs to execute the command are met
+type preReqValidator struct {
+	ctxName string
+	ns      string
+
+	k8sClientProvider    okteto.K8sClientProvider
+	oktetoClientProvider types.OktetoClientProvider
+}
+
+// newPreReqValidator returns a new preReqValidator
+func newPreReqValidator(opts ...option) *preReqValidator {
+	cfg := defaultPreReqCfg()
+	for _, opt := range opts {
+		opt(cfg)
+	}
+	return &preReqValidator{
+		ctxName:              cfg.ctxName,
+		ns:                   cfg.ns,
+		k8sClientProvider:    cfg.k8sClientProvider,
+		oktetoClientProvider: cfg.oktetoClientProvider,
+	}
+}
+
+// Validate validates that all the pre-reqs to execute the command are met
+func (v *preReqValidator) Validate(ctx context.Context) error {
+	oktetoLog.Info("validating pre-reqs")
```

Comment:
```suggestion
	oktetoLog.Info("validating pre-reqs for kubetoken")
```

could we add a little bit more context at this log? what are pre-reqs for?